File: SwipeRefresh.java
public class MainActivity extends AppCompatActivity {

    SwipeRefreshLayout swipeRefresh;
    TextView tvNumber;
    int number = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        swipeRefresh = findViewById(R.id.swipeRefresh);
        tvNumber = findViewById(R.id.number);

        swipeRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                number++;
                tvNumber.setText(String.valueOf(number));
                swipeRefresh.setRefreshing(false);
            }
        });
    }

}

File: Swipe refresh.kt
class MainActivity : AppCompatActivity() {

    private lateinit var swipeRefresh: SwipeRefreshLayout
    private lateinit var tvNumber: TextView
    var number = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        swipeRefresh = findViewById(R.id.swipeRefresh)
        tvNumber = findViewById(R.id.number)

        swipeRefresh.setOnRefreshListener {
            number++
            tvNumber.text = number.toString()
            swipeRefresh.setRefreshing(false)
        }
    }
}
